dttools/src/disk_allocator.c

```diff
@@ -10,16 +10,53 @@ See the file COPYING for details.
 #include <inttypes.h>
 #include <unistd.h>
 #include <string.h>
-
+#include "getopt.h"
 #include "disk_alloc.h"
 #include "stringtools.h"
+#include "cctools.h"
+
+static void show_help(const char *cmd) {
+
+	fprintf(stdout, "Use: %s [options]\n", cmd);
+	fprintf(stdout, "Where options are:\n");
+	fprintf(stdout, " %-30s This message\n", "-h,--help=<flag>");
+	fprintf(stdout, " %-30s Version\n", "-v,--version");
+	fprintf(stdout, " %-30s Loop device allocation\n", "create <target directory> <size (i.e. 100MB)> <filesystem>");
+	fprintf(stdout, " %-30s Loop device deallocation\n", "delete <target directory>");
+	fprintf(stdout, "\n");
+	return;
+}
 
 int main(int argc, char *argv[]) {
 
 	char *func = argv[1];
 	char *loc = argv[2];
-	char *fs = argv[3];
+	char *fs = argv[4];
```

Comment:
These array accesses are not safe without checking argc.

The right thing to do is let option parsing happen first, then pull extra arguments. Look at `man getopt_long` and the `optind` variable. Also look at Parrot:

https://github.com/cooperative-computing-lab/cctools/blob/master/parrot/src/pfs_main.cc#L1142